File: WebApplication/ClientApp/src/app/_orchestration/getBreadcrumb/eventsBreadcrumbResultDto.ts
import { DisplayNameWithIdDto } from "src/app/_orchestration/displayNameWithIdDto";
import { BreadcrumbResultDto } from "./breadcrumbResultDto";

export interface EventsBreadcrumbResultDto extends BreadcrumbResultDto {
  raceSeriesDisplayWithId?: DisplayNameWithIdDto | null
  raceDisplayWithId?: DisplayNameWithIdDto | null
  courseDisplayWithId?: DisplayNameWithIdDto | null
  irpDisplayWithId?: DisplayNameWithIdDto | null
}

File: Orchestration/GetBreadcrumb/EventsBreadcrumbResultDto.cs
namespace Orchestration.GetBreadcrumb;

public record EventsBreadcrumbResultDto : BreadcrumbResultDto
{
    public DisplayNameWithIdDto? RaceSeriesDisplayWithId { get; }

    public DisplayNameWithIdDto? RaceDisplayWithId { get; }

    public DisplayNameWithIdDto? CourseDisplayWithId { get; }

    public DisplayNameWithIdDto? IrpDisplayWithId { get; }

    public EventsBreadcrumbResultDto
    (
        LocationInfoWithUrl locationInfoWithUrl,
        DisplayNameWithIdDto? raceSeriesDisplayWithId = null,
        DisplayNameWithIdDto? raceDisplayWithId = null,
        DisplayNameWithIdDto? courseDisplayWithId = null,
        DisplayNameWithIdDto? irpDisplayWithId = null
    ) : base(locationInfoWithUrl)
    {
        RaceSeriesDisplayWithId = raceSeriesDisplayWithId;
        RaceDisplayWithId = raceDisplayWithId;
        CourseDisplayWithId = courseDisplayWithId;
        IrpDisplayWithId = irpDisplayWithId;
    }
}
